src/t8_forest_catalyst.cxx

```diff
@@ -0,0 +1,455 @@
+ * We differentiate between scalar and vector data.
+ */
+
+/* We set the input data and write the vtu files. */
+  pwriterObj->SetInputData (unstructuredGrid);
+  pwriterObj->Update ();
+  if (pwriterObj->Write ()) {
+    /* Writing failed */
+    freturn = 1;
+  }
+  else {
+    t8_errorf ("Error when writing vtk file.\n");
+  }
+
+/* We have to free the allocated memory for the cellTypes Array and the other arrays we allocated memory for. */
+/* Return whether writing was successful */
+  return freturn;
+}
+/* *INDENT-OFF* */
+vtkSmartPointer < vtkUnstructuredGrid >
+t8_build_vtk_unstructured_grid (t8_forest_t forest, int write_treeid,
+                                int write_mpirank, int write_level,
+                                int write_element_id, int curved_flag,
+                                int num_data, t8_vtk_data_field_t * data)
+{
+
+  /*Check assertions: forest and fileprefix are not NULL and forest is commited */
+  T8_ASSERT (forest != NULL);
+  T8_ASSERT (forest->rc.refcount > 0);
+  T8_ASSERT (forest->committed);
+
+  long int            point_id = 0;     /* The id of the point in the points Object. */
+  t8_locidx_t         ielement; /* The iterator over elements in a tree. */
+  t8_locidx_t         itree, ivertex;
+  double              coordinates[3];
+  double              vertex_coords[3] = { 0, 0, 0 };
+  int                 elem_id = 0;
+  t8_locidx_t         num_elements;
+  t8_gloidx_t         gtreeid;
+  t8_cmesh_t          cmesh;
+  int                 num_corners;
+
+/* Since we want to use different element types and a points Array and cellArray 
+ * we have to declare these vtk objects. The cellArray stores the Elements.
+ * The points and cellArray are needed to store the data we want to write in the Unstructured Grid. 
+ */
+  vtkNew < vtkPoints > points;
+  vtkNew < vtkCellArray > cellArray;
+  vtkNew < vtkHexahedron > hexa;
+  vtkNew < vtkVertex > vertex;
+  vtkNew < vtkLine > line;
+  vtkNew < vtkQuad > quad;
+  vtkNew < vtkTriangle > tri;
+  vtkNew < vtkWedge > prism;
+  vtkNew < vtkTetra > tet;
+
+  vtkNew < vtkQuadraticEdge > quadraticedge;
+  vtkNew < vtkQuadraticTriangle > quadratictri;
+  vtkNew < vtkQuadraticQuad > quadraticquad;
+  vtkNew < vtkQuadraticTetra > quadratictet;
+  vtkNew < vtkQuadraticHexahedron > quadratichexa;
+  vtkNew < vtkQuadraticWedge > quadraticprism;
+
+  /* 
+   * The cellTypes Array stores the element types as integers(see vtk doc).
+   */
+  num_elements = t8_forest_get_local_num_elements (forest);
+  int                *cellTypes = T8_ALLOC (int, num_elements);
+
+  /*
+   * We need the vertex coords array to be of the 
+   * correct dim. Since it is always the same
+   * in one mesh, we take the dim of one element.
+   * We add 1 if we look at a vertext(dim=0) because 
+   * an array of size 0 is not allowed. 
+   * Then we allocate memory, because we do not know
+   * beforehand how many entries the array needs.
```

Comment:
I think the code concerning this comment is deleted or elsewhere. Move or delete this comment.